### ISSUE TITLE
cannot extend table defined with literal

### ISSUE THREAD

**Issue by lewis6991** [CppCXY/emmylua-analyzer-rust#264]
![Image](https://github.com/user-attachments/assets/fa29f328-a79a-427b-bf6e-4845e4d42261)

```lua
local a = { 'a' }
a[#a + 1] = 'b'
```

`a` is being given type `['a']`, which is correct, but it is intended to be used as a list. Perhaps this can be special cased and coerced to `string[]` is needed?

Workaround is to use `@type`:

```lua
local a = { 'a' } --- @type string[]
a[#a + 1] = 'b'
```

**Comment by lewis6991** [comment]
Yes, and it's correct, but perhaps we can coerce it to an array if needed?

**Comment by xuhuanzy** [comment]
Now he shouldn't be reporting any errors, right?

**Comment by xuhuanzy** [comment]
```lua
local a = { 'a' } --- @type { 'a' } -- explicit type
```
Syntax for tuples is not yet supported; this requires a decision from @CppCXY.

**Comment by xuhuanzy** [comment]
Currently, it is not possible to provide diagnostics for tuples because it is difficult to distinguish whether they are explicitly declared tuples. Earlier discussions in cppcxy rejected removing the default parsing as tuples.

**Comment by xuhuanzy** [comment]
Now it will report an error for manually declared `---@type` `Tuple`.
